src/java/htsjdk/tribble/AbstractFeatureCodec.java

```diff
@@ -26,7 +26,10 @@
 import java.io.IOException;
 
 /**
- * Simple basic class providing much of the basic functionality of codecs
+ * Simple basic class providing much of the basic functionality of codecs.
+ * Every concrete subclass must implement {@link FeatureCodec#canDecode(String)} to indicate whether it can decode the file.
+ * Note that that method is the only way that the right codec for a file is identified and that <bold>only one</bold> codec
+ * is allowed to identify itself as being able to decode any given file.
```

Comment:
Add a comment that checking the file's extension is a perfectly acceptable implementation of `canDecode()`, and that file contents only rarely need to be checked (this should probably go in the javadoc for `canDecode()` itself in the `FeatureCodec` interface rather than here).